Dockerfile

```diff
@@ -1,13 +1,50 @@
-FROM balenalib/intel-nuc-node:10.16.3-stretch-build-20190920
-
-# Install PIP, Robot Framework, Resin-cli and Etcher-cli
-RUN apt-get update && apt-get install -y qemu-system-x86 rsync qemu-kvm minicom libftdi-dev python-pip python-setuptools python-wheel systemd && \
-    rm -rf /var/lib/apt/lists/* && \
-    pip install robotframework==3.0 requests==2.4.3 robotframework-requests==0.4.5 pylibftdi==0.15.0 && \
-    npm install --global balena-cli@10.17.6 && \
-    git clone --depth 1  --branch v1.5.57 https://github.com/resin-io/etcher.git && cd /etcher && \
-    npm install --production && \
-    ln -sf /etcher/bin/etcher /usr/local/bin/etcher
+# build
+FROM balenalib/intel-nuc-node:10-stretch-build AS builder
+
+WORKDIR /opt
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+      build-essential flex bison rsync minicom libftdi-dev python-pip \
+      python-setuptools python-wheel systemd && \
+      rm -rf /var/lib/apt/lists/*
+
+ENV VIRTUAL_ENV=/opt/venv
+
+RUN pip install virtualenv && python -m virtualenv $VIRTUAL_ENV
+
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+COPY requirements.txt .
+
+RUN pip install -r requirements.txt
+
+# https://github.com/nodejs/node/issues/19348
+RUN pip install -r requirements.txt && \
+      npm install --global balena-cli@10.17.6 && \
+      wget -q https://download.qemu.org/qemu-4.2.0.tar.xz && tar -xf qemu-4.2.0.tar.xz && \
+      cd qemu-4.2.0 && ./configure && make && make install
+
+
+# runtime
+FROM balenalib/intel-nuc-node:10-stretch-run
+
+ENV VIRTUAL_ENV=/opt/venv
+
+ENV PATH="$VIRTUAL_ENV/bin:/usr/local/bin:$PATH"
+
+ENV PYTHONHOME=$VIRTUAL_ENV
+
+COPY --from=builder /opt/venv /opt/venv
+
+COPY --from=builder /usr/lib/python2.7 /usr/lib/python2.7
+
+COPY --from=builder /usr/local /usr/local
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
```

Comment:
moving this RUN before the `COPY` statements will lead to much better layer caching since this apt-get is going to be updated less frequently than the other artifacts